components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/execution/PreIssueAccessTokenRequestBuilder.java

```diff
@@ -112,12 +120,57 @@ private Event getEvent(OAuthTokenReqMessageContext tokenMessageContext, Map<Stri
             eventBuilder.userStore(new UserStore(authorizedUser.getUserStoreDomain()));
         }
 
-        eventBuilder.accessToken(getAccessToken(tokenMessageContext, claimsToAdd));
+        // Retrieving app information here to reduce the perf overhead.
+        OAuthAppDO oAuthAppDO;
+        try {
+            oAuthAppDO = getAppInformation(tokenMessageContext);
+        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+            throw new ActionExecutionRequestBuilderException(
+                    "Failed to generate pre issue access token action request for application: " +
+                            tokenMessageContext.getOauth2AccessTokenReqDTO().getClientId() + " grant type: " +
+                            tokenMessageContext.getOauth2AccessTokenReqDTO().getGrantType(), e);
+        }
```

Comment:
Move this to a method for readability

Reply:
Fixed 49caebcb5cca897a855b8ca05cb5db3e23952e9d